src/runtime/capabilities.ts

```diff
@@ -479,6 +479,16 @@ export class Capabilities {
     return new Capabilities(ranges);
   }
 
+  setCapability(capability: Capability, type: typeof Capability): boolean {
```

Comment:
this is really dumb, but i cannot find a way to get `typeof` out of the Capability object :(

Reply:
The only thing I can think of is to use a tag for all Capability subtypes. e.g. something like this:

```
type CapabilityTag = 'persistence' | 'ttl' | ...;

class Capability {
  abstract readonly tag: CapabilityTag;
}

class Persistence extends Capability {
  tag: 'persistence';
  ...
}

  setCapability(capability: Capability) {
     let existing = this.getWithTag(capablity.tag);
     ...
  }
```

Reply:
i really like this, done. thanks!

Reply:
I'll merge now, but let me know if you have comments and i'll address them in a follow up PR.

Reply:
LGTM!